packages/@o3r/design/src/core/design-token/renderers/design-token-style.renderer.ts

```diff
@@ -98,6 +98,31 @@ export const getTokenSorterByRef: DesignTokenListTransform = (variableSet) => {
   };
 };
 
+/**
+ * Reorganize the Tokens based on a ordered list of RegExps
+ * @param regExps Ordered list of regular expressions defining the order of the Tokens.
+ */
+export const getTokenSorterFromRegExpList: (regExps: RegExp[]) => DesignTokenListTransform = (regExps) => (_variableSet, options) => {
+
+  return (tokens) =>
+    tokens
+      .map((token) => ({ index: regExps.findIndex((r) => r.test(token.getKey(options?.tokenVariableNameRenderer))), token }))
```

Comment:
We should validate that the regexp does not have a `g` flag
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#using_test_on_a_regex_with_the_global_flag
no ?

Reply:
I think it is up to the user to care about it.
We can add a word regarding it in the doc, but I don't think you should forbid it

Reply:
I agree to add a message in the doc 
but I will at least raised a warning because the behavior is totally uncontrol in this case (I would prefer that we throw an error, I don't see use case where it could have a sense)
If in the code we have `token.getKey(options?.tokenVariableNameRenderer).match(r)`
the user will not have the problem and it's not their role to check how we coded it.